### DISCUSSION TITLE
pip install location

### DISCUSSION THREAD

**Discussion by berensn** [vkbo/novelWriter#744] [Help] [answered]
When using pip to install, what is the install location?
On Windows I can see {user}/appdata/local/novelwriter and {user}/appdata/roaming/novelwriter
The {user}/appdata/local/novelwriter only has a .conf file and the other path is empty
The pip and python folders in appdata are also empty and there's nothing in Program Files/Program Files (x86)

**Comment by vkbo** [comment]
You'd have to check pip's documentation, ~~but I'd guess it would be `{user}/appdata/local/novelwriter`~~. I don't think it uses Program Files, although I'm not much of a Windows user so I'm not sure. I don't control the `pip` command's behaviour from my end in any case.

In addition to what pip does, novelWriter finds a location for the config and additional app data from the Qt `QStandardPaths` class. Specifically, `ConfigLocation` and `AppDataLocation`. These are OS dependant, and are described here: https://doc.qt.io/qt-5/qstandardpaths.html

**Reply by vkbo** [reply]
After some further digging, all I could find is this:
`C:\Users\{USER}\AppData\Local\Programs\Python\Python38\Lib\site-packages`

**Comment by berensn** [comment, marked as answer]
I found it, thanks to your help.
actual location of assets, files, etc is: Users{user}\AppData\Local\Programs\Python\Python38-32\Lib\site-packages\nw
location of the exe is: Users{user}\AppData\Local\Programs\Python\Python38-32\Scripts\novelWriter.exe

There is also Users{user}\AppData\Local\Programs\Python\Python38-32\Lib\site-packages\novelWriter-1.2.2.dist-info where the RECORD file gave the name of the folder, nw, where the assets, files, etc are stored; along with SHA256 hashes of all the files, I presume for integrity checks, probably managed by pip.

**Reply by vkbo** [reply]
Yeah. all of this is handled by pip.

The locations are intended for Python packages rather than apps like novelWriter. It would be nice if pip had an option to install it as an app with a more logical location like Program Files. I used to produce exe installers, but Windows is rather alarmist when it comes to installers, and not very open source welcoming. Doing it properly costs a lot of money, so I just dropped it.

Since novelWriter doesn't require much to run from source, I provide the minimal zip packages for each OS that can just be extracted where the user wants them (for instance to Program Files). A simple `python setup.py win-install` command will create the desktop and start menu icons and file association.

**Reply by berensn** [reply]
Yeah, I went with the 'easier' method with pip, which to be fair, is ultra easy. When I install it on linux I'll go the other route
